### PR TITLE
CLDR-17111 Deprecate old style zone IDs (e.g. EST5EDT, EET)

### PR DESCRIPTION
**Pull request by justingrant**
CLDR-17111

- [ ] This PR completes the ticket.

ALLOW_MANY_COMMITS=true

This PR revises changes from #3736 (cc @yumaoka) to align CLDR's aliases for recently deprecated zones to match IANA's Link=>Zone mappings in the following recent commits:

https://github.com/eggert/tz/commit/782d082623aaa130178d944bdbfbb563d2e1adfa
https://github.com/eggert/tz/commit/a0b09c0230089252acf2eb0f1ba922e99f7f4a03

It also fixes tests broken by these deprecations.

### CODE REVIEW

**Comment by yumaoka** [comment]
The changes look fine. All changes are aligned to what IANA TZ database maintainer agreed with.
Because you already have the correct changes, I'll close my PRs previously created.

For ICU testing - we'll find out if there are any changes required on ICU side after this is merged. I don't think CLDR test fails with this change.

**Review by yumaoka — APPROVED** [review]
Confirmed all changes are aligned to what IANA TZ database maintainer agreed with.

**Comment by justingrant** [comment]
> The changes look fine. All changes are aligned to what IANA TZ database maintainer agreed with. Because you already have the correct changes, I'll close my PRs previously created.

Great!  Glad we were able to resolve this upstream. Easier for everyone. 

> For ICU testing - we'll find out if there are any changes required on ICU side after this is merged. I don't think CLDR test fails with this change.

Yep, fixing the CLDR tests was thankfully easy: it only required adding EST5EDT, etc. zones to the deprecated-zone exception list in TestLocale.java. I can confirm that tests failed before this test-code change and passed afterwards.

**Comment by yumaoka** [comment]
@justingrant Are you waiting for someone to merge this? There is an issues in this PR. First, you have a commit with a comment not starting with `CLDR-17111`. Can you fix this? It is not a requirement to squash changes into one, but for this change, it's probably a good idea to do so.

If you are not able to do this, I can create a separate PR and merge it. I cannot modify the branch, so that's only the solution if you're not going to correct PR issues and merge it.

**Comment by justingrant** [comment]
> @justingrant Are you waiting for someone to merge this? 

Yep!  Also, I don't have permissions to run the GH build and test actions so someone will need to run those. 

> you have a commit with a comment not starting with `CLDR-17111`

Fixed!

> not a requirement to squash changes into one, but for this change, it's probably a good idea to do so.

Sqaushed and rebased to main. 

Let me know if there's anything more that I need to do. Thanks @yumaoka!

**Pull request merged**
